### PR TITLE
chore: replace travis with circleci

### PR DESCRIPTION
**Pull request by ted123**
Closes S-04888

### CODE REVIEW

**Comment by fakewaffle** [comment]
You'll probably need to use node 12.x.

**Comment by ted123** [comment]
@fakewaffle can you grant me rebuild and ssh build access for this commit in circle ci?

**Comment by fakewaffle** [comment]
That's not a permission I see that I can grant?

**Comment by ted123** [comment]
@fakewaffle I notified it to the circle team and they are looking into it. The test already passed in circle ci so I think this commit is ready for merging.

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "RPC library for AMQP protocol",
   "main": "index.js",
   "scripts": {
-    "test": "gulp test && gulp inspect-queues",
```
Comment (fakewaffle) [inline package.json]: Why was `&& gulp inspect-queues` removed?

@yamii, We're still using that, aren't we?

File: circle.yml [hunk 2]
```diff
@@ -1,29 +1,16 @@
...
+  version: 0.12.0
+ services:
+  - docker
+dependencies:
+ override:
   - npm install
-
-script:
-
+  - docker run -d --name rabbitmq rabbitmq:3.5-management
```
Comment (openam) [inline circle.yml]: @fakewaffle noticed you're not binding the ports to the host when starting this up,

You probably also need to add a `hosts` entry to have `rabbitmq` go to the right spot. https://circleci.com/docs/configuration#hosts

Comment (ted123) [inline circle.yml]: Thanks. Its working now.

**Comment by gideonairex** [comment]
:+1: Im good with this. Any thoughts @fakewaffle ?

**Pull request merged**